cluster/saltbase/salt/docker/init.sls

```diff
@@ -114,6 +114,36 @@ lxc-docker-{{ override_docker_ver }}:
       - lxc-docker-{{ override_docker_ver }}: /var/cache/docker-install/{{ override_deb }}
 {% endif %}
 
+# Default docker systemd unit file doesn't use an EnvironmentFile; replace it with one that does.
+{% if pillar.get('is_systemd') %}
+
+{{ pillar.get('systemd_system_path') }}/docker.service:
+  file.managed:
+    - source: salt://docker/docker.service
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - defaults:
+        environment_file: {{ environment_file }}
+
+# The docker service.running block below doesn't work reliably
```

Comment:
The root issue that caused me to just move this out of salt is this one: https://github.com/saltstack/salt/issues/16778

Salt won't do a daemon-reload out-of-the-box, by design.  This was not what I was expecting.

I can definitely add the extra watch, but it won't work if e.g. docker has already been loaded by systemd.